dispatch.go

```diff
@@ -40,13 +40,20 @@ func (d *dispatch) Call(ctx context.Context, name string, arg interface{}, res i
 	profiler := d.log.StartProfiler("call %s", name)
 	defer profiler.Stop()
 
+	select {
+	case <-d.stopCh:
```

Comment:
That's fine, but now the regression test you added for the cancellation hang bug (TestDispatchCallAfterClose) now tests something else trivial. So if you really want to add this, you'll have to add another test that actually tests what TestDispatchCallAfterClose used to test (i.e., an RPC call makes it past this select, but then Close() is called, and it shouldn't hang).

Also, if you're gonna do this here, you probably should do it for Notify, too. And add similar tests, etc. (I notice there are none for Notify).

Seems better to just keep this PR to just fixing the bug, and not adding any more code paths.